### DISCUSSION TITLE
include entt.hpp in two module cause a crash

### DISCUSSION THREAD

**Discussion by armorhu** [skypjack/entt#807] [General]
here is my problem。
In my project，I have two module，both want use same entt_registry.
So I share the same entt::registry instance with them.
But , when i add component i found type_id of diffecnent component will be same.

in module a
registry->emplace<CompA> , 
const auto index = type_seq<Component>::value();
index = 0;

in module b
registry->emplace<CompB> , 
const auto index = type_seq<Component>::value();
index = 0; // same as module a

and i think this is because 
![image](https://user-images.githubusercontent.com/4246258/146130680-5fc67601-348e-4c6c-af27-616d21cd6712.png)

both module has it's own static varible.

so what can i do...

**Comment by skypjack** [comment]
You've a few options:
* Wait until v3.9, it solves this problem.
* Use `master` (not the single include file), this problem is already solved upstream.
* Generate your own single include file from `master` (see `scripts/amalgamate.py`).

**Reply by armorhu** [reply]
I saw v3.9 changelog post here
https://github.com/skypjack/entt/discussions/740.
which not mention this improve。

and mater's version is v3.8?

**Reply by skypjack** [reply]
> EnTT across boundaries
> TODO

😂 let me fix it, you're right. This is just a changelog preview but yeah, I should keep it in sync until it's official.
`master` is not v3.8, the `v3.8` tag is v3.8 and the `v3.8.x` branch contains the last version for this release, that is `v3.8.1`.
`master` is `master`, it contains all the changes that are meant to be part of the upcoming v3.9 at the moment. 🤷‍♂️

**Reply by armorhu** [reply]
i try to Generate your own single include file from master. and get this error . What's wrong with it?
![image](https://user-images.githubusercontent.com/4246258/146148404-e40a27f4-9a42-4dc0-88e0-72ae09538f91.png)

**Reply by armorhu** [reply]
my python version is 2.7.0
![image](https://user-images.githubusercontent.com/4246258/146149753-d8c9a270-acde-4f9a-954f-e29e8167b325.png)

**Reply by skypjack** [reply]
I think you need python3 actually.
